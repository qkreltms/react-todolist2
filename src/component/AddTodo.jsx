import React, { Component } from 'react';
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import { withStyles } from '@material-ui/core/styles';

const styles = (theme) => ({
    root: {
      padding: '2px 4px',
      display: 'flex',
      alignItems: 'center',
      width: 400,
      margin: 'auto',
    },
    input: {
      marginLeft: 8,
      flex: 1,
    }
})

class AddTodo extends Component {
    state = {
        content: ''
    }

    handleOnSubmit = (e) => {
        e.preventDefault()
        this.props.addTodo(this.state.content)
        this.setState({
            content: '' 
        })
    }

    setContent = (e) => {
        this.setState({
            content: e.target.value 
        })
    }
    
    render() {
        const { classes } = this.props
        return (
            <form onSubmit={this.handleOnSubmit}>
                <Paper elevation={1} className={classes.root}>
                    <FormattedMessage id='add-todo-here'>
                        {msg => (
                            <InputBase 
                            className={classes.input}
                            placeholder={msg}
                            type='text' 
                            onChange={this.setContent} 
                            value={this.state.content}
                            />                    
                        )}
                    </FormattedMessage>
                </Paper>
            </form>
        );
    }
}

AddTodo.propTypes = {
    addTodo: PropTypes.func,
    classes: PropTypes.object.isRequired
}

AddTodo.defaultTypes = {
    addTodo: () => console.warn('addTodo is not defined'),
    classes: () => console.warn('classes is not defined')
}

export default withStyles(styles)(AddTodo)