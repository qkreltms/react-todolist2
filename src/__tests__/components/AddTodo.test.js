import React from 'react';
import AddTodo from '../../component/AddTodo';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('AboutTobo', () => {
    test('renders without crashing', () => {
        const component = shallow(<AddTodo />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
      })
})
