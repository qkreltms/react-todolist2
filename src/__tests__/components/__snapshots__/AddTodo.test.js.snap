// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutTobo renders without crashing 1`] = `
<AddTodo
  classes={
    Object {
      "input": "AddTodo-input-2",
      "root": "AddTodo-root-1",
    }
  }
/>
`;
